functions/ocr/app/test/index.test.js

```diff
@@ -1,438 +1,271 @@
+      file,
       vision: visionMock,
       translate: translateMock,
-      topic: topic
+      topic
     }
   };
 }
 
-function getMockContext () {
-  return {
-    done: sinon.stub(),
-    success: sinon.stub(),
-    failure: sinon.stub()
-  };
-}
-
-describe('functions:ocr', function () {
-  it('processImage does nothing on delete', function () {
-    var context = getMockContext();
-
-    getSample().sample.processImage(context, {
-      timeDeleted: 1234
-    });
-
-    assert.equal(context.done.calledOnce, true);
-    assert.equal(context.failure.called, false);
-    assert.equal(context.success.called, false);
+describe(`functions:ocr`, () => {
+  it(`processImage does nothing on delete`, () => {
+    return getSample().program.processImage({ data: { resourceState: `not_exists` } });
   });
 
-  it('processImage fails without a bucket', function () {
-    var expectedMsg = 'Bucket not provided. Make sure you have a ' +
-      '"bucket" property in your request';
-    var context = getMockContext();
+  it(`processImage fails without a bucket`, () => {
+    const error = new Error(`Bucket not provided. Make sure you have a "bucket" property in your request`);
 
-    getSample().sample.processImage(context, {});
-
-    assert.equal(context.failure.calledOnce, true);
-    assert.equal(context.failure.firstCall.args[0], expectedMsg);
-    assert.equal(context.success.called, false);
-    assert.equal(console.error.called, true);
+    return getSample().program.processImage({ data: {} })
+      .catch((err) => {
+        assert.deepEqual(err, error);
+      });
   });
 
-  it('processImage fails without a name', function () {
-    var expectedMsg = 'Filename not provided. Make sure you have a ' +
-      '"name" property in your request';
-    var context = getMockContext();
-
-    getSample().sample.processImage(context, {
-      bucket: bucket
-    });
+  it(`processImage fails without a name`, () => {
+    const error = new Error(`Filename not provided. Make sure you have a "name" property in your request`);
 
-    assert.equal(context.failure.calledOnce, true);
-    assert.equal(context.failure.firstCall.args[0], expectedMsg);
-    assert.equal(context.success.called, false);
-    assert.equal(console.error.called, true);
+    return getSample().program.processImage({ data: { bucket: bucketName } })
+      .catch((err) => {
+        assert.deepEqual(err, error);
+      });
   });
 
-  it('processImage handles detectText error', function (done) {
-    var expectedMsg = 'error';
-    var context = {
-      success: assert.fail,
-      failure: function () {
-        assert.equal(context.failure.calledOnce, true);
-        assert.equal(context.failure.firstCall.args[0], expectedMsg);
-        assert.equal(context.success.called, false);
-        assert.equal(console.error.calledWith(expectedMsg), true);
-        done();
+  it(`processImage processes an image`, () => {
+    const event = {
+      data: {
+        bucket: bucketName,
```

Comment:
No, as I want it to be clear that the variable represents a string. `bucket` is the variable name I would assign for an actual bucket object.